### ISSUE TITLE
'Connection refused' in RunFromNetwork on iOS

### ISSUE THREAD

**Issue by tilk48** [chgibb/hydro-sdk#41]
I started to experience this weird error when trying to load a file with the RunFromNetwork widget.

`SocketException (SocketException: OS Error: Connection refused, errno = 61, address = localhost, port = 50250)`

I tried my own url as well as the one from the example git.
This happens on the iOS simulator as well as on a real device but only in debug mode. It works fine if compiled in release mode.

The strange thing is that it worked fine a hours ago, can you reproduce it on your side?

**Comment by chgibb** [comment]
@tilk48 I'm having trouble reproducing this under `example-project`. Are you seeing this error when trying to load a local file or one from the internet?

**Comment by tilk48** [comment]
You are right! The problem seems to have been on my side. The simulator could not reach the internet for some reason, restarting it fixed the issue. We should maybe add a better error description for the not so unlikely case of being unable to reach the provided url.

**Comment by chgibb** [comment]
@tilk48 I definitely agree about error messages.

**Comment by tilk48** [comment]
Ive been working on increasing the stability and error resistance of the `RunFromNetwork` widget because it continued to be somewhat unreliable for me. I found that the provided url is being overwritten with another localhost url in this section of the code (runFromNetwork.dart, line 81) :

```dart
_debugUrl = kDebugMode && Platform.isAndroid
        ? "http://10.0.2.2:5000"
        : kDebugMode && Platform.isIOS ? "http://localhost:5000" : "";
```

I removed this and refactored a few other parts of the code and it seems like the stability has increased. So my question is why this was in there in the first place?
